### PR TITLE
Upgrade integration testing

### PR DESCRIPTION
**Pull request by woksin**
With this code The Aviator is using an unpublished version of the integration testing SDK image. We need to publish it when this is pulled in.

* Use the latest rc versions of the SDK
* Make EventsController methods awaitable by having them return a Task
* Added methods for committing multiple events
* Use EventObject type alias that represents the content the Events
* Comment out tests for checking the position of the stream processors as it is checking the volatile state of stream processor states. We could do something that bases of the "current state", but that creates a coupling between the test scenarios. I'm not sure what we want to do here, that's why I commented it out.
* Modified the unstable event horizon scenario to considering a limitation in the Runtime's event horizon subscription system

### CODE REVIEW

**Comment by woksin** [comment]
Converted to draft as I'm fixing more things

File: Source/TheAviator/gherkin/ScenarioEnvironment.ts [hunk 1]
```diff
@@ -12,47 +12,58 @@ import { Scenario } from './Scenario';
...
 
     async dumpEventStore(scenario: Scenario) {
         this.forEachMicroservice(async (microservice) => {
             const microserviceDestinationDirectory = this._flightPaths.forMicroserviceInScenario(scenario, microservice);
-            const destinationDirectory = path.join(microserviceDestinationDirectory, 'eventStore');
-
-            if (!fs.existsSync(destinationDirectory)) {
-                fs.mkdirSync(destinationDirectory, { recursive: true });
+            const destinationDirectory = path.join(microserviceDestinationDirectory, eventStoreDumpFolderName);
+            const directoryExists = await promisify(fs.exists)(destinationDirectory);
```
Comment (einari) [inline Source/TheAviator/gherkin/ScenarioEnvironment.ts]: FS has a promises object on it with async promise based versions. So you can skip the promisify all together.
(https://nodejs.org/api/fs.html#fs_fs_promises_api)

```suggestion
            const directoryExists = await fs.promises.exists(destinationDirectory);
```

File: Source/TheAviator/gherkin/ScenarioEnvironment.ts [hunk 2]
```diff
@@ -12,47 +12,58 @@ import { Scenario } from './Scenario';
...
+            if (!directoryExists) {
+                await promisify(fs.mkdir)(destinationDirectory, { recursive: true });
```
Comment (einari) [inline Source/TheAviator/gherkin/ScenarioEnvironment.ts]:
```suggestion
                await fs.promises.mkdir(destinationDirectory, { recursive: true });
```

**Review by einari — CHANGES_REQUESTED** [review]
Looks good.
I would suggest using the `.promises` object on the FS object instead of the `promisify` approach, since its already promisified.

**Comment by woksin** [comment]
Nice, I did not know that!

File: Source/TheAviator/gherkin/ScenarioEnvironment.ts [hunk 3]
```diff
@@ -12,47 +12,58 @@ import { Scenario } from './Scenario';
...
             const microserviceDestinationDirectory = this._flightPaths.forMicroserviceInScenario(scenario, microservice);
-            const destinationDirectory = path.join(microserviceDestinationDirectory, 'eventStore');
-
-            if (!fs.existsSync(destinationDirectory)) {
-                fs.mkdirSync(destinationDirectory, { recursive: true });
+            const destinationDirectory = path.join(microserviceDestinationDirectory, eventStoreDumpFolderName);
+            const directoryExists = await fs.promises.exists(destinationDirectory);
+            if (!directoryExists) {
+                await fs.promises.mkdir(destinationDirectory, { recursive: true });
             }
```
Comment (joelhoisko) [inline Source/TheAviator/gherkin/ScenarioEnvironment.ts]: `fs.exists()` has been [deprecated](https://nodejs.org/api/fs.html#fs_fs_exists_path_callback) so we could just use `fs.promises.mkdir()` straight up. 

With the `recursive: true` it [won't raise an error:](https://nodejs.org/api/fs.html#fs_fs_mkdir_path_options_callback) 

>  Calling fs.mkdir() when path is a directory that exists results in an error only when recursive is false.

So we can just call it straight up or we could use `fs.stat()` or `fs.access()` on it first, but both of these methods [docs](https://nodejs.org/api/fs.html#fs_fs_stat_path_options_callback) advise against it:

> Using fs.stat() to check for the existence of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended. Instead, user code should open/read/write the file directly and handle the error raised if the file is not available.

```suggestion
            await fs.promises.mkdir(destinationDirectory, { recursive: true });
```

**Comment by joelhoisko** [comment]
All working now 👍

**Pull request merged**